### PR TITLE
Measure connection latency

### PR DESCRIPTION
**Pull request by lbbrhzn**

### CODE REVIEW

**Review by drc38 — CHANGES_REQUESTED** [review]
Looks good, if you add the measurement state class then statistics over time can be generated to see how stable the latency is

**Comment by lbbrhzn** [comment]
> Looks good, if you add the measurement state class then statistics over time can be generated to see how stable the latency is

i'll do, thanks!

**Comment by lbbrhzn** [comment]
Hmm, what SensorDeviceClass should I use? There is only timestamp, no timedelta.

**Comment by drc38** [comment]
> Hmm, what SensorDeviceClass should I use? There is only timestamp, no timedelta.

Does it need one? Set to None?

File: custom_components/ocpp/api.py [hunk 1]
```diff
@@ -774,6 +777,31 @@ async def _get_specific_response(self, unique_id, timeout):
...
+    async def measure_connection_latency(self):
+        """Measure the connection latency."""
+        try:
+            while True:
+                t0 = time.perf_counter()
+                pong_waiter = await self._connection.ping()
+                await pong_waiter
+                t1 = time.perf_counter()
+                latency = round(1000 * (t1 - t0))
+                _LOGGER.info(
```
Comment (drc38) [inline custom_components/ocpp/api.py:789]: suggest changing this to debug or commenting out before merging

File: custom_components/ocpp/api.py [hunk 2]
```diff
@@ -774,6 +777,31 @@ async def _get_specific_response(self, unique_id, timeout):
 
         return resp
 
+    async def measure_connection_latency(self):
+        """Measure the connection latency."""
+        try:
+            while True:
+                t0 = time.perf_counter()
+                pong_waiter = await self._connection.ping()
```
Comment (drc38) [inline custom_components/ocpp/api.py:785]: use asyncio.wait_for(self._connection.ping(), timeout=20) to ensure the loop does not get stuck if there is no response to the ping

**Pull request merged**